src/gui/src/dbDescriptors.cpp

```diff
@@ -230,6 +230,206 @@ static odb::dbTechLayer* getLayerSelection(odb::dbTech* tech,
 
 ////////
 
+DbTechDescriptor::DbTechDescriptor(odb::dbDatabase* db) : db_(db)
+{
+}
+
+std::string DbTechDescriptor::getName(std::any object) const
+{
+  return "Default";
+}
+
+std::string DbTechDescriptor::getTypeName() const
+{
+  return "Tech";
+}
+
+bool DbTechDescriptor::getBBox(std::any object, odb::Rect& bbox) const
+{
+  return false;
+}
+
+void DbTechDescriptor::highlight(std::any object, Painter& painter) const
+{
+}
+
+Descriptor::Properties DbTechDescriptor::getProperties(std::any object) const
```

Comment:
Please add getManufacturingGrid()  (check hasManufacturingGrid first), getNonDefaultRules(), and getMetalWidthViaMap()

Reply:
Ideally all rules would have a getTech() method.  The db at this point is used with a single tech so I guess nobody bothered.  We may have to change that in the future but for now you can just the the tech from the db.

dbMetalWidthViaMap is usually a table in LEF (dummy example):
```
 LIBRARY LEF58_METALWIDTHVIAMAP STRING "
  METALWIDTHVIAMAP
   VIA   v1    1.000              1.000               VIA11
   VIA   v1    1.000              2.000               VIA12
   VIA   v1    1.000              3.000               VIA13
   VIA   v1    2.000              1.000               VIA21
   VIA   v1    2.000              2.000               VIA22
   VIA   v1    2.000              3.000               VIA23
```
If we could display it that way it would be nice.

Reply:
I'm sorry, i'm a bit confused. In your example there are two values per line, but, as far as I understood there are four values that can be returned from dbMetalWidthViaMap:

below_layer_width_low
below_layer_width_high
above_layer_width_low
above_layer_width_high

I was trying to print the values so I could understand better but, do the technologies in the test folder have a metal width via map?. I've run different ones and I got an empty map when calling `tech->getMetalWidthViaMap()`. Maybe I'm doing something wrong.